### PR TITLE
*: Use Bind/Get global sysvar accessor instead of type assertion

### PR DESCRIPTION
**Pull request by shenli**
@siddontang

### CODE REVIEW

File: sessionctx/variable/sysvar.go [hunk 1]
```diff
@@ -581,3 +581,27 @@ type GlobalSysVarAccessor interface {
 	// SetGlobalSysVar sets the global system variable name to value.
 	SetGlobalSysVar(ctx context.Context, name string, value string) error
```
Comment (siddontang) [inline sessionctx/variable/sysvar.go]: The struct name is GlobalSysVarAccessor which indicates that we will handle global sys var, so I thin function Set/Get is enough.

Comment (shenli) [inline sessionctx/variable/sysvar.go]: This is implemented in session. It has many functions. So get/set is too simple.

Comment (siddontang) [inline sessionctx/variable/sysvar.go]: got it.

File: expression/variable.go [hunk 2]
```diff
@@ -82,7 +83,7 @@ func (v *Variable) Eval(ctx context.Context, args map[interface{}]interface{}) (
 			return value, nil
 		}
 	}
-	value, err := ctx.(variable.GlobalSysVarAccessor).GetGlobalSysVar(ctx, name)
+	value, err := globalVars.GetGlobalSysVar(ctx, name)
```
Comment (siddontang) [inline expression/variable.go]: why not using variable.GetGlobalSysVar instead?

Comment (shenli) [inline expression/variable.go]: In that way, I have to define two more functions to wrap GetGlobalSysVar and SetGlobalSysVar.

Comment (siddontang) [inline expression/variable.go]: got it.

**Comment by siddontang** [comment]
LGTM

File: sessionctx/variable/sysvar.go [hunk 3]
```diff
@@ -581,3 +581,27 @@ type GlobalSysVarAccessor interface {
...
 }
+
+// keyType is a dummy type to avoid naming collision in context.
+type globalSysVarAccessorKeyType int
```
Comment (qiuyesuifeng) [inline sessionctx/variable/sysvar.go]: globalSysVarAccessorKeyType

Comment (shenli) [inline sessionctx/variable/sysvar.go]: PTAL

**Pull request merged**